src/components/withDelayToggleButtonState.js

```diff
@@ -0,0 +1,80 @@
+import React, {Component} from 'react';
+import PropTypes from 'prop-types';
+import getComponentDisplayName from '../libs/getComponentDisplayName';
+
+const withDelayToggleButtonStatePropTypes = {
+    /** A value whether the button state is complete */
+    isButtonStateComplete: PropTypes.bool.isRequired,
+
+    /** A function to call to change the complete state */
+    toggleButtonStateComplete: PropTypes.func.isRequired,
```

Comment:
NAB: how about this rename?

`toggleButtonStateComplete` -> `toggleDelayButtonState`